### ISSUE TITLE
odd vision behaviour when light is limited in global illumination w/o darkvision

### ISSUE THREAD

**Issue by rapus95** [dev7355608/limits#16]
situation: Global Illumination enabled, Darkness 1
![image](https://github.com/dev7355608/limits/assets/5960902/9a91b310-6cd9-435d-9eb4-926ea4399a38)
and that's the configuration for the drawn area
![image](https://github.com/dev7355608/limits/assets/5960902/2410fba7-9ded-44a3-956e-29ae89d57a34)

the character doesn't have darkvision. But the suddenly hidden creature is less than 10 ft from the border. So why does it become invisible?
![image](https://github.com/dev7355608/limits/assets/5960902/1704ef36-7f99-4dcf-8f3b-926c0a94dd7b)
disabling the light restriction makes the creature visible again, but then, there's no light at all, darkness 1, why does that even make a difference when less than 10 ft in the area?

I've been thinking if this is the same issue as #3 but came to to the conclusion that it's somewhat like the reverse issue.

EDIT: I begin to assume that #3 actually isn't a bug but correctly designed. And the fix to it (global illumination doesn't reveal tokens anymore) is the reason for the bug in this issue. Because as seen in my image, pure vanilla global illumination, even when in Darkness 1, reveals tokens.

**Comment by rapus95** [comment]
that this behaviour is incorrect can be seen here:
![image](https://github.com/dev7355608/limits/assets/5960902/e0dd56b6-7c3a-4f1c-b134-6c309ca33722)
For that, I enclosed the whole map into a limits area which limits "Light" to 1000ft. Now, creatures without darkvision cannot see anymore even with global illumination on and darkness to 0. Generally this can be really nice since it allows for being able to see the whole map w/o unveiling any tokes, but when it's day (global illumination) that doesn't make much sense.

**Comment by dev7355608** [comment]
The problem is that the global light doesn't have an origin. I currently make the assumption that the rays of the global light come from an origin infinitely far way. Also the the limiting regions do not have a bounded top/bottom: they extend infinitely below/above, meaning any limit set on light will stop the light rays of the global rays far, far above the ground. So what you seeing is actually expected behavior. But perhaps I should just revert the change I did to address #3 and let the global light ignore limitations considering there's no way to change the top/bottom elevations of limiting regions at the moment and the global light doesn't have to be sunlight necessarily; it could be something like cave illuminated by glowing mushrooms for example, where the light is emitted essentially by every point in the scene, in which case ignoring light limits is actually the correct behavior.

Limiting light doesn't limit the range at which light can be seen. So if you limit light to 1000 ft., it won't effect any light source with a range below 1000 ft. You probably want to limit _Light Perception_ instead, which controls the max range at which tokens still can see light. Limiting light only affects the area the light source covers; it does not affect the distance at which light an be seen.

**Comment by rapus95** [comment]
> But perhaps I should just revert the change I did to address https://github.com/dev7355608/limits/issues/3 and let the global light ignore limitations considering there's no way to change the top/bottom elevations of limiting regions at the moment and the global light doesn't have to be sunlight necessarily;

I second that, postpone a proper solution for #3 until foundry v12 when https://github.com/foundryvtt/foundryvtt/issues/3078 hopefully lands (+ many of those in https://github.com/foundryvtt/foundryvtt/issues/10104 to introduce some sort of height/level into core foundry). Right now, I'd suggest assuming ambient light to behave like "light at every point in scene" and consider parallel light rays (infinitely far away large source) as "misuse of global illumination" as for "cave illuminated by glowing mushrooms". Alternatively, if you're really invested to solve it before v12 lands, you can work on compatibility with [Elevated Vision](https://foundryvtt.com/packages/elevatedvision) and [Wall Height](https://foundryvtt.com/packages/wall-height)+Levels since these introduce a matter of height/elevation even for terrain and far away light source.

> Limiting light doesn't limit the range at which light can be seen. So if you limit light to 1000 ft., it won't effect any light source with a range below 1000 ft. You probably want to limit Light Perception instead, which controls the max range at which tokens still can see light. Limiting light only affects the area the light source covers; it does not affect the distance at which light an be seen.

Yes I know that, that's what I used (and that's the reason the shadows in the lower right have this fancy outline which perfectly mirrors the outline of the forest in the images of my first post). And that's what I suspected to be the primary misunderstanding in #3.

**Comment by dev7355608** [comment]
I think this was fixed or working correctly. The global light doesn't have an origin (or every point in the scene is an origin). Note core global light is not (visually) blocked by roof tiles (or shouldn't be, there's a bug in V12).
